### PR TITLE
qt5: add some platforms plugins

### LINKED ISSUE #5868: [package] qt/6.1.1: Platform plugins should not be linked unless Default_IF is true
with `cmake_find_multi`

on windows the following plugins are built (`QWindowsIntegrationPlugin` and `QWindowsDirect2DPlugin`)

both are platform plugins and duplicate symbols by default platform plugins are disabled but only enabled / added when the DEFAULT_IF argument is set to true (thus in this case since it's a windows platform only `QWindowsIntegrationPlugin` should be built/linked.

### LINKED ISSUE #6227: [package] qt/6.1.2: static build missing windows platform plugin
### Package and Environment Details
  * Package Name/Version: **qt/6.1.2**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC 19.29**
  * Conan version: **conan 1.38.0**
  * Python version: **Python 3.9.5**

### Conan profile
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
compiler.runtime=MDd
build_type=Debug
[options]
[conf]
[build_requires]
[env]
```

### Description

Building a simple "hello world" program using Qt6::Widgets fails to run when linking the static qt/6.1.2 libraries:

```
qt.qpa.plugin: Could not find the Qt platform plugin "windows" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```
Using Qt6's own generated cmake targets, these plugins are linked automatically when consuming e.g. Qt6::Widgets.

I've tried a few things to try to resolve this manually,
* linking the Qt6::qtQWindowsIntegrationPlugin target explicitly -- no change
* using `qt6_import_plugins(example INCLUDE qwindows)` -- results in 
  ```
  Warning: plug-in qwindows is not known to the current Qt installation
  ```
* importing the plugin from a source file:
  ```cpp
  #include <QtPlugin>

  Q_IMPORT_PLUGIN(qwindows);
  ```
  result:
  ```
  LNK2019: unresolved external symbol "struct QStaticPlugin const __cdecl qt_static_plugin_qwindows(void)
  ```
  (though I'm not sure I have the plugin name right).

### Steps to reproduce 
I've created a small repro here: <https://github.com/srnwk/conan-qt-static-repro>

Assuming the above profile is called msvc-debug, and running in a suitable vcvarsall.bat environment:
```
git clone https://github.com/srnwk/conan-qt-static-repro
cd conan-qt-static-repro
conan install . -if build/conan -pr msvc-debug
cmake -G Ninja -B build -DCMAKE_BUILD_TYPE=Debug
ninja -C build
build/example.exe
```

### LINKED ISSUE #6318: [package] qt/5.15.2: Could not find the Qt platform plugin "xcb" in ""
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **qt/5.15.2**
  * Operating System+version: **Linux Mint 20.2**
  * Compiler+version: **GCC 9**
  * Conan version: **conan 1.38.0**
  * Python version: **Python 3.8.10**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=9
compiler.libcxx=libstdc++11
build_type=Release
[options]
[conf]
[build_requires]
[env]

```

### Steps to reproduce (Include if Applicable)
1. Build an application that depends on qt
2. Attempt to run the application, you'll receive the error:
```
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **qt/5.15.2**

fixes #6318
fixes #6227
fixes #5868

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qt/5.x.x/conanfile.py [hunk 1]
```diff
@@ -256,6 +256,7 @@ def _enablemodule(mod):
                 _enablemodule(module)
 
     def validate(self):
+        tools.check_min_cppstd(self, "11")
```
Comment (prince-chrismc) [inline recipes/qt/5.x.x/conanfile.py:259]: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-is-a-toolscheck_min_cppstd-call-not-enough

File: recipes/qt/5.x.x/conanfile.py [hunk 2]
```diff
@@ -883,6 +888,30 @@ def _create_plugin(pluginname, libname, type, requires):
...
                 gui_reqs.append("libjpeg::libjpeg")
             _create_module("Gui", gui_reqs)
+
+            if self.settings.os == "Windows":
+                _create_plugin("QWindowsIntegrationPlugin", "qwindows", "platforms", ["Core", "Gui"])
+                self.cpp_info.components["qtQWindowsIntegrationPlugin"].system_libs = ["advapi32", "dwmapi", "gdi32", "imm32",
+                    "ole32", "oleaut32", "shell32", "shlwapi", "user32", "winmm", "winspool", "wtsapi32"]
+            elif self.settings.os == "Android":
+                _create_plugin("QAndroidIntegrationPlugin", "qtforandroid", "platforms", ["Core", "Gui"])
+                self.cpp_info.components["qtQWindowsIntegrationPlugin"].system_libs = ["android", "jnigraphics"]
```
Comment (prince-chrismc) [inline recipes/qt/5.x.x/conanfile.py:898]: typo?

```suggestion
                self.cpp_info.components["qtQAndroidIntegrationPlugin"].system_libs = ["android", "jnigraphics"]
```

**Review by prince-chrismc — COMMENTED** [review]
Two minor things

File: recipes/qt/5.x.x/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/qt/5.x.x/conanfile.py:259]: check_min_cppstd is already guarded against empty cppstd: https://github.com/conan-io/conan/blob/f230bf104842c160be6ea377497151fe1ea64c38/conans/client/tools/settings.py#L41 and in this case it effectively uses the actual compiler's default for std.
If we add this `if self.settings.compiler.get_safe("cppstd")`, then the users using default configuration without explicit cppstd setting (most users) will not be stopped by this error, and then they get a cryptic error message when they try to consume the project. I had to help several users with this situation.

Comment (prince-chrismc) [reply recipes/qt/5.x.x/conanfile.py:259]: We had previously sided that we should block those unsupported by default compilers.
I agree about the behavior but perhaps we should go about it differently?

Comment (ericLemanissier) [reply recipes/qt/5.x.x/conanfile.py]: OK, this is controversial, and I don't want to delay the integration of the other changes, so here we go

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Pull request merged**